lib/netproto/netproto_connect.c

```diff
@@ -129,6 +138,14 @@ getserveraddr(void)
 
 	/* If we have a new lookup, update the cache. */
 	if (tmp_addr != NULL) {
+
+		/* First time we run getserveraddr, register atexit. */
```

Comment:
This isn't necessarily the first time we run getserveraddr.  It's the first time we have an address to cache though.